packages/lesswrong/server/callbacks/userCallbackFunctions.tsx

```diff
@@ -352,59 +357,73 @@ export async function updateDigestSubscription(data: Partial<DbUser>, {oldDocume
       throw err
     }
     data.subscribedToDigest = false
+    data.subscribedToNewsletter = false
     return data;
   }
 
-  if (
-    isAnyTest ||
-    !hasDigests ||
-    data.subscribedToDigest === undefined || // When a mutation doesn't reference subscribedToDigest
+  const noDigestUpdate = !hasDigests ||
+    data.subscribedToDigest === undefined ||
     data.subscribedToDigest === oldDocument.subscribedToDigest
-  ) {
+  const noNewsletterUpdate = !hasNewsletter ||
+    data.subscribedToNewsletter === undefined ||
+    data.subscribedToNewsletter === oldDocument.subscribedToNewsletter
+  if (isAnyTest || (noDigestUpdate && noNewsletterUpdate)) {
     return data;
   }
 
   const mailchimpAPIKey = mailchimpAPIKeySetting.get();
   const mailchimpForumDigestListId = mailchimpForumDigestListIdSetting.get();
-  if (!mailchimpAPIKey || !mailchimpForumDigestListId) {
-    return handleErrorCase("Error updating digest subscription: Mailchimp not configured")
+  const mailchimpEANewsletterListId = mailchimpEAForumNewsletterListIdSetting.get();
+  const digestUpdateButNoListId = hasDigests && !mailchimpForumDigestListId
+  const newsletterUpdateButNoListId = hasNewsletter && !mailchimpEANewsletterListId
+  if (!mailchimpAPIKey || digestUpdateButNoListId || newsletterUpdateButNoListId) {
+    return handleErrorCase("Error updating subscription: Mailchimp not configured")
   }
 
   const email = getUserEmail(newDocument)
   if (!email) {
-    return handleErrorCase(`Error updating digest subscription: no email for user ${data.displayName}`)
+    return handleErrorCase(`Error updating subscription: no email for user ${data.displayName}`)
   }
 
   const { lat: latitude, lng: longitude, known } = userGetLocation(newDocument);
-  const status = data.subscribedToDigest ? 'subscribed' : 'unsubscribed';
+  const digestStatus = data.subscribedToDigest ? 'subscribed' : 'unsubscribed';
+  const newsletterStatus = data.subscribedToNewsletter ? 'subscribed' : 'unsubscribed';
   const emailHash = md5(email!.toLowerCase());
 
-  const res = await fetch(`https://us8.api.mailchimp.com/3.0/lists/${mailchimpForumDigestListId}/members/${emailHash}`, {
-    method: 'PUT',
-    body: JSON.stringify({
-      email_address: email,
-      email_type: 'html', 
-      ...(known && {location: {
-        latitude,
-        longitude,
-      }}),
-      merge_fields: {
-        FNAME: data.displayName,
+  const updates = [
+    {noUpdate: noDigestUpdate, listId: mailchimpForumDigestListId, status: digestStatus},
+    {noUpdate: noNewsletterUpdate, listId: mailchimpEANewsletterListId, status: newsletterStatus}
+  ].filter((u) => !!u.listId)
```

Comment:
Could you just filter the noUpdate ones out here so they don't have to be skipped again below

Reply:
Obviously 🤦